docs/root/configuration/http/http_conn_man/headers.rst

```diff
@@ -615,6 +615,12 @@ Supported variable names are:
     Upstream metadata cannot be added to request headers as the upstream host has not been selected
     when custom request headers are generated.
 
+%REQUEST_METADATA(["namespace", "key", ...])%
```

Comment:
Should we rename this FILTER_METADATA as it could be response metadata set in the encode side of a filter?

Reply:
The corresponding access log formatter is called `DYNAMIC_METADATA`, so I think we should match that name. It would also be good to make this respond to the same syntax. I have it on my todo list to unify all these formatters into a single set.

Reply:
Ok

Reply:
I did the rename, however making it respond to the same syntax as this access logger formatter would make it inconsistent with `UPSTREAM_METADATA()`. So I think in this context it might be better to keep the header formatters consistent and later on unify their syntax with the access log ones given it's a breaking change?

Or were you suggesting introducing new header formatters now -- following the syntax used in the access log formatters -- with new names now?

Reply:
I'd rather both the header formatter and the access log formatter use the same syntax for the same operator name, even if that means they're inconsistent within header formatter (because I think they already are inconsistent in access log formatters).

Reply:
Ok.

Reply:
Sorry for the back and forth -- but I actually think keeping the dynamic metadata header formatter as I am proposing it might make more sense (even though I still generally see the merit in unifying the formatters):

* the access log one emits JSON (a bit annoying to have JSON in request/response headers)
* I might need it for things like hashing on the header that came from metadata (the extra quotes of generated JSON are a problem there)
* I need it to debug simple metadata values, so the extra JSON wrapping again adds very little

Hmm.